tests/test_knn_classifier.py

```diff
@@ -0,0 +1,184 @@
+        compressed_input, compressed_combined = self.model._compress_sample(
+            self.sample_input
+        )
+
+        assert isinstance(compressed_input, np.ndarray)
+        assert isinstance(compressed_combined, np.ndarray)
+        assert compressed_input.shape == compressed_combined.shape
+
+    def test__calculate_distance(self) -> None:
+        compressed_input, compressed_combined = self.model._compress_sample(
+            self.sample_input
+        )
+        distance = self.model._calculate_distance(compressed_input, compressed_combined)
+
+        assert isinstance(distance, np.ndarray)
+        assert distance.shape == compressed_input.shape == compressed_combined.shape
+
+    def test_concatenate_with_space(self) -> None:
+        a = "hello there"
+        b = "who goes there?"
+        assert self.model.concatenate_with_space(a, b) == "hello there who goes there?"
+
+    def test_predict(self) -> None:
+        top_k = 1
+        (distance, labels, similar_samples) = self.model.predict(
+            self.sample_input, top_k
+        )
+
+        assert distance.shape == (
+            len([self.sample_input]),
+            self.model.training_inputs.shape[0],
+        )
+        assert labels.shape == (top_k, len([self.sample_input]))
+        assert similar_samples.shape == (top_k, len([self.sample_input]))
+
+        test_set_size = random.randint(1, 50)
+        test_set = [self.sample_input for _ in range(test_set_size)]
+        top_k = 2
+        (distance, labels, similar_samples) = self.model.predict(test_set, top_k)
+
+        assert distance.shape == (test_set_size, self.model.training_inputs.shape[0])
+        assert labels.shape == (test_set_size, top_k)
+        assert similar_samples.shape == (test_set_size, top_k)
+
+    def test_negative_top_k(self) -> None:
+        test_set_size = random.randint(1, 50)
+        test_set = [self.sample_input for _ in range(test_set_size)]
+        top_k = -1
+
+        with pytest.raises(AssertionError):
+            self.model.predict(test_set, top_k)
+
+    def test_top_k_bigger_than_test_set(self) -> None:
+        test_set_size = random.randint(1, 10)
+        test_set = [self.sample_input for _ in range(test_set_size)]
+        top_k = test_set_size + 1
+        with pytest.raises(AssertionError):
+            self.model.predict(test_set, top_k)
+
+    def test_sampling_percentage(self) -> None:
+        test_set_size = random.randint(1, 10)
+        test_set = [self.sample_input for _ in range(test_set_size)]
+        top_k = 1
+        (full_distance, full_labels, full_similar_samples) = self.model.predict(
+            test_set, top_k, sampling_percentage=1.0
+        )
+        (distance, labels, similar_samples) = self.model.predict(
+            test_set, top_k, sampling_percentage=0.1
+        )
+
+        assert full_labels.shape == labels.shape
+        assert full_similar_samples.shape == similar_samples.shape
+
+    def test_sample_data(self):
```

Comment:
This omits a `-> None` return-type annotation, which appears unintended since every other test-case method in the test suite now has one.